system/Router/RouteCollection.php

```diff
@@ -319,6 +319,9 @@ public function loadRoutes(string $routesFile = APPPATH . 'Config/Routes.php')
             return $this;
         }
 
+        // Normalize the path string in routesFile
+        $routesFile = realpath($routesFile) ?: $routesFile;
```

Comment:
Because we installed phpstan-strict-rules, short ternary operator is not allowed.

> Disallow short ternary operator (?:) - implies weak comparison, it's recommended to use null coalesce operator (??) or ternary operator with strict condition.
https://github.com/phpstan/phpstan-strict-rules

```
 ------ ---------------------------------------------------------------------------------------------------------------------------- 
  Line   system/Router/RouteCollection.php                                                                                           
 ------ ---------------------------------------------------------------------------------------------------------------------------- 
  305    Ignored error pattern #^Short ternary operator is not allowed\. Use null coalesce operator if applicable or consider using  
         long ternary\.$# in path /Users/kenji/work/codeigniter/official/CodeIgniter4/system/Router/RouteCollection.php is expected  
         to occur 2 times, but occurred 3 times.                                                                                     
  1687   Short ternary operator is not allowed. Use null coalesce operator if applicable or consider using long ternary.             
 ------ ---------------------------------------------------------------------------------------------------------------------------- 
```

Reply:
In this case, fix will be like this. But it is a bit difficult to read.
@paulbalandan What do you think?

```diff
--- a/system/Router/RouteCollection.php
+++ b/system/Router/RouteCollection.php
@@ -320,7 +320,7 @@ class RouteCollection implements RouteCollectionInterface
         }
 
         // Normalize the path string in routesFile
-        $routesFile = realpath($routesFile) ?: $routesFile;
+        $routesFile = (realpath($routesFile) === false) ? $routesFile : realpath($routesFile);
 
         // Include the passed in routesFile if it doesn't exist.
         // Only keeping that around for BC purposes for now.
```

Reply:
My take on internal functions that may return false is to throw on these instances. However, we test on vfs so it may not be okay. Also, we should not double call realpath.

```php
$routesFile = ($tmp = realpath($routesFile)) !== false ? $tmp : $routesFile;

```

Although I'm not a fan of assignment in if conditions.

Reply:
Yes, we have that code because we may use vfs.
How about this for the time being?

```diff
--- a/system/Router/RouteCollection.php
+++ b/system/Router/RouteCollection.php
@@ -320,7 +320,8 @@ class RouteCollection implements RouteCollectionInterface
         }
 
         // Normalize the path string in routesFile
-        $routesFile = realpath($routesFile) ?: $routesFile;
+        $realpath   = realpath($routesFile);
+        $routesFile = ($realpath === false) ? $routesFile : $realpath;
 
         // Include the passed in routesFile if it doesn't exist.
         // Only keeping that around for BC purposes for now.
```

Reply:
@kenjis  So in that case line 305:

```
$this->routeFiles[$routeKey] = realpath($routesFile) ?: $routesFile;
```

and line 1687:

```
$verb = $verb ?: $this->getHTTPVerb();
```

should also be fixed ?

Reply:
The fix should be on a separate PR.

Reply:
Late to the game here and not trying to start a war but: I'm okay with short-form ternary when dealing with clearly-defined return types.